demo/src/main/java/com/example/demo/web/PersonController.java

```diff
@@ -1,58 +1,88 @@
 package com.example.demo.web;
 
+import com.example.demo.model.ContactMethod;
 import com.example.demo.model.Person;
-
+import com.example.demo.model.PostalAddress;
 import com.example.demo.service.PersonService;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import javax.validation.Valid;
 import java.util.List;
+import java.util.Optional;
 
 
 @RestController
 @RequestMapping("/person")
 public class PersonController {
 
     private final PersonService personService;
+
     public PersonController(PersonService personService) {
         this.personService = personService;
     }
 
     @PostMapping("/create")
-    public ResponseEntity<Person> createPerson(@RequestBody Person person) {
-        Person person1 = this.personService.createPerson(person);
-        return new ResponseEntity<>(person1, HttpStatus.CREATED);
+    public ResponseEntity<Person> createPerson(@Valid @RequestBody Person person) {
+        //Optional<Person> person1 = this.personService.createPerson(person);
+        //return new ResponseEntity<>(person1, HttpStatus.CREATED);
+        return this.personService.createPerson(person)
+                .map(p -> ResponseEntity.ok().body(p))
+                .orElseGet(() -> ResponseEntity.badRequest().build());
+//        for(PostalAddress address: person.getAddresses())
+//            address.setPerson(person);
+//        return this.personService.createPerson(person);
     }
 
     @GetMapping("/listAll")
     public ResponseEntity<List<Person>> getAllPersons() {
         List<Person> personList = this.personService.listAll();
-        return new ResponseEntity<>(personList,HttpStatus.OK);
+        return new ResponseEntity<>(personList, HttpStatus.OK);
     }
+
     @GetMapping("/{id}")
     public ResponseEntity<Person> getPersonById(@PathVariable Long id) {
-        Person person1 = this.personService.getbyID(id);
-        return new ResponseEntity<>(person1,HttpStatus.OK);
+        Person person = this.personService.getbyID(id);
+        return new ResponseEntity<>(person, HttpStatus.OK);
     }
 
     @GetMapping("/email")
     public ResponseEntity<List<Person>> getPersonByEmail(@RequestParam String email) {
-        List<Person> persons = this.personService.findByEmail(email)
-                ;
+        List<Person> persons = this.personService.findByEmail(email);
         return ResponseEntity.ok().body(persons);
     }
 
     @GetMapping("/streetadd")
-    public ResponseEntity<List<Person>> getByStreetAddress(@RequestParam String streetAddress){
-    List<Person> personList = this.personService.findByStreetAddress(streetAddress);
-    return ResponseEntity.ok().body(personList);
+    public ResponseEntity<List<Person>> getByStreetAddress(@RequestParam String streetAddress) {
```

Comment:
Only one Person should be returned (if unique) and please correct to camelCase everywhere